### DISCUSSION TITLE
Can we have a jupyter notebook and a Python interactive window use the same kernel?

### DISCUSSION THREAD

**Discussion by Kameshkk** [microsoft/vscode-jupyter#12162] [Questions and Answers]
I like to use Jupyter notebooks, and I also like to have a IPython console (Python interactive window in vscode) open alongside the notebook so that I can rapidly inspect variables or run snippets of code. Is it possible to open a Python interactive window in vscode and have it connect to the python kernel of a jupyter notebook that is already open inside vscode? Appreciate any ideas for workarounds!
Thanks.

**Comment by crazy4pi314** [comment]
For quick inspection of variables in the current notebook session, I find the [variable explorer](https://code.visualstudio.com/docs/datascience/jupyter-notebooks#_variable-explorer-and-data-viewer) to be super helpful :)

**Comment by faultdiagnosistoolbox** [comment]
I open up a terminal (in vscode or outside) and run
% jupyter console --existing
and it attaches to the existing running server. I find it work fine that way.

**Comment by DonJayamanne** [comment]
> t variables or run snippets of code. Is it possible to open a Python interactive window in vscode and have it connect to the python kernel of a jupyter notebook that is already open inside vscode? 

This suggestion is tracked here https://github.com/microsoft/vscode-jupyter/issues/4543
